### PR TITLE
fix: convertPointToView() to apply ScrollView contentOffset

### PR DESCRIPTION
**Pull request by jquick-axway**
**JIRA:**
https://jira.appcelerator.org/browse/TIMOB-28398

**Summary:**
If calling `childView.convertPointToView(point, scrollView)`, when the destination is a `ScrollView`, then its `contentOffset` should be applied to the returned value. This would make it easier to scroll to a child view within a scroll view.

**Test:**
_(Run this test on both Android and iOS.)_
1. Build and run `ScrollViewOffsetVerticalTest.js` attached to [TIMOB-28398](https://jira.appcelerator.org/browse/TIMOB-28398).
2. Tap on the "Navigate" button in the bottom right corner.
3. Select "Blue View" in the dialog.
4. Verify the top of the blue view is scrolled to the top.
5. Tap on the "Navigate" button and select "Blue View" again.
6. Verify the top of the blue view is still displayed. It should not scroll at all. _(This was a bug.)_
7. Tap on the "Navigate" button and select "Red View".
8. Verify you are scrolled to the top of the red view.

### CODE REVIEW

**Review by garymathews — APPROVED** [review]
Android CR: **PASS**

File: iphone/TitaniumKit/TitaniumKit/Sources/API/TiViewProxy.m [hunk 1]
```diff
@@ -768,6 +773,7 @@ - (TiPoint *)convertPointToView:(id)args
   TiPoint *tiPoint = [[TiPoint alloc] autorelease];
   [tiPoint setX:NUMFLOAT(convertDipToDefaultUnit(pointOffsetDips.x + givenPoint.x))];
   [tiPoint setY:NUMFLOAT(convertDipToDefaultUnit(pointOffsetDips.y + givenPoint.y))];
+  [tiPoint add:[arg2 contentOffset]];
```
Comment (vijaysingh-axway) [inline iphone/TitaniumKit/TitaniumKit/Sources/API/TiViewProxy.m:776]: I think we can replace above line with following code and all other changes can be discarded.
```suggestion
   if ([arg2 respondsToSelector:@selector(contentOffset)]) {
    // In scrollview contentOffset should be applied to returned value
    TiPoint *contentOffset = [arg2 contentOffset];
    [tiPoint setX:NUMFLOAT(convertDipToDefaultUnit(pointOffsetDips.x + contentOffset.point.x + givenPoint.x))];
    [tiPoint setY:NUMFLOAT(convertDipToDefaultUnit(pointOffsetDips.y + contentOffset.point.y + givenPoint.y))];
  } else {
    [tiPoint setX:NUMFLOAT(convertDipToDefaultUnit(pointOffsetDips.x + givenPoint.x))];
    [tiPoint setY:NUMFLOAT(convertDipToDefaultUnit(pointOffsetDips.y + givenPoint.y))];
  }
```

Comment (vijaysingh-axway) [reply iphone/TitaniumKit/TitaniumKit/Sources/API/TiViewProxy.m:776]: Do we need to update doc as well?

Comment (vijaysingh-axway) [inline iphone/TitaniumKit/TitaniumKit/Sources/API/TiViewProxy.m:776]: I think we can replace above 3 lines with following code and all other changes can be discarded.
```suggestion
   if ([arg2 respondsToSelector:@selector(contentOffset)]) {
    // In scrollview contentOffset should be applied to returned value
    TiPoint *contentOffset = [arg2 contentOffset];
    [tiPoint setX:NUMFLOAT(convertDipToDefaultUnit(pointOffsetDips.x + contentOffset.point.x + givenPoint.x))];
    [tiPoint setY:NUMFLOAT(convertDipToDefaultUnit(pointOffsetDips.y + contentOffset.point.y + givenPoint.y))];
  } else {
    [tiPoint setX:NUMFLOAT(convertDipToDefaultUnit(pointOffsetDips.x + givenPoint.x))];
    [tiPoint setY:NUMFLOAT(convertDipToDefaultUnit(pointOffsetDips.y + givenPoint.y))];
  }
```

**Comment by lokeshchdhry** [comment]
FR Passed.
Checked on Android & IOS.

**Pull request merged**